pkg/server/server_entrypoint_tcp.go

```diff
@@ -171,6 +171,21 @@ func (e *TCPEntryPoint) StartTCP(ctx context.Context) {
 		}
 
 		safe.Go(func() {
+			// Set read/write for respect timeout even without first byte.
```

Comment:
```suggestion
			// Enforce read/write timeouts at the connection level so that they get
			// triggered even without a first byte being seen, as opposed to with
			// timeouts at the server level.
```